misc/hooks/pre-commit-makerst

```diff
@@ -2,4 +2,4 @@
 
 # Git pre-commit hook that checks the class reference syntax using makerst.py.
 
-doc/tools/makerst.py doc/classes modules --dry-run
+python doc/tools/makerst.py doc/classes modules --dry-run
```

Comment:
That will fail on environments where `python` is Python 2.

Does it work with `./doc/tools/makerst.py doc/classes modules --dry-run`?

Reply:
No, same error with `./doc/tools/makerst.py doc/classes modules --dry-run`

Reply:
Ah it's infuriating that Python packaging is still a mess after 12 years of Python 3...

By adding `python` here you're simply overriding the shebang `#!/usr/bin/env python3` which doesn't work for you as apparently the Windows distribution of Python 3 doesn't install any `python3` target, only `python`. So it would be better to just change the shebang to `#!/usr/bin/env python`.

But most current Linux distros still have `python` as a symlink to `python2`, while newer ones have `python` as a symlink to `python3`.

Reply:
Hacky for hacky, I guess you can do:
```bash
PYTHON=python
py_ver=$($PYTHON -c "import sys; print(sys.version_info.major)")
if [[ "$py_ver" != "3" ]]; then
  PYTHON+=3
fi

$PYTHON doc/tools/makerst.py doc/classes modules --dry-run
```

Reply:
Yes, it seems it's working too differently on windows to make it work correctly without a hack.

This command actually works directly in a console:
`"doc/tools/makerst.py" doc/classes modules --dry-run`
Because of the .py file association with the python launcher.

But it looks like the file association doesn't work anymore when the same command is launched from the pre-commit script :/